File: selenium-webdriver-demo-javascript/remote-demo.js
const args = require('argumently');
const port = args.has('-p') ? args.get('-p') : 4444; 

const { Builder, Capabilities } = require("selenium-webdriver");
const chrome = require('selenium-webdriver/chrome');
var capabilities = Capabilities.chrome();
var options = new chrome.Options();
options.addArguments('--start-maximized');

(async function helloSelenium() {
    let driver = new Builder()
        .usingServer("http://localhost:" + port)
        .forBrowser('chrome')
        .setChromeOptions(options)
        .withCapabilities(capabilities)
        .build();
    try {
        await driver.get('http://www.google.com');
        sleep(16000);
    } finally {
        await driver.quit();
    }
})();  

function sleep(milliseconds) {
    const date = Date.now();
    let currentDate = null;
    do {
      currentDate = Date.now();
    } while (currentDate - date < milliseconds);
  }
  

File: selenium-webdriver-demo-python/remote-demo.py
from selenium import webdriver
import time
import sys

print(sys.argv)
port = sys.argv[2] if sys.argv[1] == '-p' else '4444'

options = webdriver.ChromeOptions()
# https://stackoverflow.com/a/12213723/552792
options.add_argument("--start-maximized")
driver = webdriver.Remote(
    command_executor='http://localhost:'+port,
    options=options
)
driver.get("http://www.google.com")
time.sleep(10)
driver.quit() 
